### PR TITLE
Update functions with MT repo and create constants file

### LINKED ISSUE #245: Updating Roadchoice Config and Common Functions
### Summary

-Update common block functions to be up to date with VT & MT
-Redirect imports related to new functions
-Enable constants.json file in order to set configs from sharepoint

### Priority

Blocker

### Description

Roadchoice is currently out of date on functionality updates. These updates need to be made before any RC blocks are further developed so that for example V2 blocks can be developed and compatible with RC. 

### Value creation

Maintainability & scalability

### Cost of fixing

Low cost

### Technical complexity

S

### References

#235 Blocked by this ticket

### PR DESCRIPTION
**Pull request by santi-homps**
Fix #245 and #240

Test URLs:
- Before: https://main--vg-roadchoice-com--hlxsites.hlx.page/
- After: https://245-update-rc--vg-roadchoice-com--hlxsites.hlx.page/

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -15,8 +15,27 @@ import {
...
+  addFavIcon,
+  createElement,
+  decorateIcons,
+  getPlaceholders,
+  loadDelayed,
+  slugify,
+  variantsClassesToBEM,
+} from './common.js';
+
+// import {
```
Comment (jonatan-lledo-netcentric) [inline scripts/scripts.js:31]: if this is not needed, please, remove

File: scripts/scripts.js [hunk 2]
```diff
@@ -209,6 +263,15 @@ function buildSearchForm(main, head) {
   main.prepend(section);
 }
 
+function buildSubNavigation(main, head) {
```
Comment (jonatan-lledo-netcentric) [inline scripts/scripts.js:266]: this function is needed in this brand?

File: scripts/scripts.js [hunk 3]
```diff
@@ -229,6 +293,12 @@ export function decorateLinks(block) {
   [...block.querySelectorAll('a')]
     .filter(({ href }) => !!href)
     .forEach((link) => {
+      /* eslint-disable no-use-before-define */
+      // if (isVideoLink(link)) {
```
Comment (jonatan-lledo-netcentric) [inline scripts/scripts.js:297]: same as before, if is not needed, please remove

File: scripts/scripts.js [hunk 4]
```diff
@@ -247,24 +317,132 @@ export function decorateLinks(block) {
...
       const url = new URL(link.href);
       const external = !url.host.match('roadchoice.com') && !url.host.match('.hlx.(page|live)') && !url.host.match('localhost');
-      if (url.pathname.endsWith('.pdf') || external) {
+      if (url.host.match('build.roadchoice.com') || url.pathname.endsWith('.pdf') || external) {
         link.target = '_blank';
       }
     });
 }
 
+function decorateSectionBackgrounds(main) {
```
Comment (jonatan-lledo-netcentric) [inline scripts/scripts.js:326]: this function needs to be reviewed because is using MT's color scheme

File: scripts/scripts.js [hunk 5]
```diff
@@ -442,12 +549,11 @@ export function loadScriptIfNotLoadedYet(url, attrs) {
 }
 
 /**
- * Creates a new block element with the specified name and content, and loads it into the page.
+ *
```
Comment (jonatan-lledo-netcentric) [inline scripts/scripts.js:552]: why is needed remove extra JSDoc explanation ?

File: scripts/scripts.js [hunk 6]
```diff
@@ -541,12 +647,9 @@ export function input(...items) { return domEl('input', ...items); }
...
 export function button(...items) { return domEl('button', ...items); }
 
-/**
- * A helper function that delays the execution of a function
- * @param {function} func the function to execute
- * @param {number} timeout the timeout in milliseconds, 200 by default
- * @returns {function} the function that will be executed after the timeout
- */
+/* Helper for delaying something like
+takes function as argument, default timout = 200
```
Comment (jonatan-lledo-netcentric) [inline scripts/scripts.js:651]: why remove JSDoc extra info?

**Review by jonatan-lledo-netcentric — COMMENTED** [review]
in general is ok, but there are som commented code and JSDoc comments that has been removed that needs to take a look

File: scripts/scripts.js (hunk 4 above)
Comment (santi-homps) [reply scripts/scripts.js:326]: yes, this is yet not applied but if we are going to add the same functionalities than on MT or VT we might need it. I'll change the colors.

File: scripts/scripts.js (hunk 6 above)
Comment (santi-homps) [reply scripts/scripts.js:651]: not intentional, just that we should have all the same in all 3 brands I believe. I'll add it back

File: scripts/scripts.js (hunk 2 above)
Comment (santi-homps) [reply scripts/scripts.js:266]: maybe not for now, but as we intend to equalize the functionalities all around I guess we can keep it there for now.

File: scripts/video-helper.js [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+  const banner = createElement('div', { classes: 'low-resolution-banner' });
+  banner.innerHTML = `${lowResolutionMessage} <button class="low-resolution-banner-cookie-settings">${changeCookieSettings}</button>`;
+  banner.querySelector('button').addEventListener('click', () => {
+    window.OneTrust.ToggleInfoDisplay();
+  });
+
+  return banner;
+}
+
+// export function showVideoModal(linkUrl) {
```
[the remark this answers is not in the document]
Comment (santi-homps) [reply scripts/video-helper.js:92]: I can deleted and we'll re copy it again in case we need it in the future

**Review by jonatan-lledo-netcentric — APPROVED** [review]

**Review by jonatan-lledo-netcentric — APPROVED** [review]

**Review by alexiscoelho — APPROVED** [review]

**Pull request merged**